modules/lambda/variables.tf

```diff
@@ -266,12 +267,6 @@ variable "iam_policy_description" {
   default     = "Minimum SSM read permissions for Lambda IAM Role"
 }
 
-variable "policy_json" {
-  type        = string
-  description = "IAM policy to attach to the Lambda IAM role"
-  default     = null
-}
-
```

Comment:
Ah what was the reasoning behind JSON > YAML policies? I'm surprised as sticking with YAML support for policies seems logical since JSON policies inside Stacks feels like we're mixing and matching our formats in an ugly way. It seems inconsistent in the components as there is a [JSON policy variable input in the kms component](https://github.com/cloudposse/terraform-aws-components/blob/main/modules/kms/README.md#input_policy), but [the `s3-bucket` component has an policy statements input as a map](https://github.com/cloudposse/terraform-aws-components/tree/main/modules/s3-bucket#input_iam_policy_statements).

Reply:
JSON is the native format if IAM policies, it is what is published in AWS documentation, and it is output by Terraform AWS data sources and resources. Practically no one uses YAML.

We have old code that uses old policy modules that takes Terraform objects and converts them to policies, but what that entails is us developing an alternate specification language for policies when we already have to deal with 2: JSON and Terraform. So we are trying to move away from that and instead only specify policies in JSON or Terraform.

Reply:
That's interesting. I also thought exposing the statements to the yaml interface was convenient and handy. I recall a client doing that and another client asking for it which led to creating a generic interface for https://github.com/cloudposse/terraform-aws-iam-policy in order to expose policy statements as a generic any map.

In ref to maybe old code (in addition to matts references), we currently have some components that expose iam policy statements as an input that can be fed via yaml.

https://github.com/cloudposse/terraform-aws-components/tree/main/modules/ecs-service#input_iam_policy_statements

There may be other components too

https://github.com/search?q=repo%3Acloudposse%2Fterraform-aws-components%20iam_policy_statements&type=code

I think the efs controller has a recent addition to its iam policy statement resource file too

Reply:
@Nuru I've added back JSON policy support, so it's possible to set one of them. Please review if that fits better the community best practices. Thank you!

Reply:
I think @gberenice's support for both looks good, but I question if we want to do that. I would double down on wanting to use YAML > JSON in components. All good if that is not what Cloud Posse wants to support, but I'd reiterate that it just feels wrong to be using JSON blocks within our YAML when the two are so portable between one another and whenever we're including custom policies inside Atmos Stack YAMLs, we're doing so because they require customization from standard AWS IAM Policies 90% of the time (otherwise, they'd likely just be built into the component).

Reply:
It should be noted that as implemented (lists), it doesn't support the finer points of atmos's ability to deepmerge. I understand why - since the interface was copied from how it already works in JSON. Just point it out, as a consideration.

Reply:
@Nuru I've utilized [terraform-aws-iam-policy](https://github.com/cloudposse/terraform-aws-iam-policy) in my recent changes. That also allows us to avoid creating the excessive (in this case) `aws_iam_policy` resource. Please let me know your thoughts.